src/main/java/de/szut/lf8_project/lf8/project/ProjectService.java

```diff
@@ -28,6 +31,20 @@ public ProjectEntity readById(long id) {
         }
         return queryedProject.get();
     }
+    public void deleteProjectById(long projectId) {
+        Optional<ProjectEntity> projectToDelete = repository.findById(projectId);
+
+        if (projectToDelete.isPresent()) {
+            ProjectEntity project = projectToDelete.get();
+            if (project.getProjectLeader().equals(true)) {
+                repository.delete(project);
+            } else {
+                throw new InvalidDataException("401, Sie sind nicht autorisiert, das Projekt zu löschen");
+            }
+        } else {
+            throw new InvalidDataException("404, Projekt nicht gefunden");
```

Comment:
The inclusion of the error code is not needed as the Exception deals with that automatically.